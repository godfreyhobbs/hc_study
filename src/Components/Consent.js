import * as PropTypes from "prop-types";
import React from "react";

function Consent(props) {
  return (<div>
    <button onClick={props.consent}>Digitally Sign Consent</button><h1>
    <a target="_blank" href="./consent.txt">Print or download the consent form for your records</a></h1>
  </div>);
}

Consent.propTypes = { consent: PropTypes.func };

export default Consent;
