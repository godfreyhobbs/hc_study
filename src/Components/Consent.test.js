import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Consent from "./Consent";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Consent  />, div);
  ReactDOM.unmountComponentAtNode(div);
});


it('renders correctly', () => {
  const tree = renderer
    .create(<Consent />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
