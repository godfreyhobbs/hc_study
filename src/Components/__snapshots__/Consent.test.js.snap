// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    onClick={undefined}
  >
    Digitally Sign Consent
  </button>
  <h1>
    <a
      href="./consent.txt"
      target="_blank"
    >
      Print or download the consent form for your records
    </a>
  </h1>
</div>
`;
